client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -185,6 +186,29 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        ReaderGroupConfig oldConfig = synchronizer.getState().getConfig();
+        Set<Stream> newStreams = config.getStartingStreamCuts().keySet();
+        if (oldConfig != null) {
+            Set<Stream> oldStreams = oldConfig.getStartingStreamCuts().keySet();
+            // If it were a subscriber, unsubscribe from all old streams
+            if (oldConfig.isSubscriber()) {
+                oldStreams.forEach(s -> getAndHandleExceptions(controller.deleteSubscriber(scope, s.getStreamName(), groupName)
```

Comment:
Instead of deleting old Streams and then adding new Streams, it may be good to get the differential set (old-new) Streams and (new-old) Stream and act on just those, to avoid unnecessary network calls.

Reply:
I have used differential sets instead.